### ISSUE TITLE
[Recurbate] HTTP Error 403 / "This video is only available for registered users"

### ISSUE THREAD

**Issue by capybaraclique** [yt-dlp/yt-dlp#7912]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [x] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Europe

### Provide a description that is worded well enough to be understood

When trying to run yt-dlp with Recurbate I always get a 403 error. I'm able to play the video in my browser using the same IP with absolutely no problems. Downloading the video via direct url using the developer console also works.
Even when providing the correct login data via cookies.txt the extractor doesn't seem to work properly.

I've tried:
- different IPs / proxies
- different URLs / videos
- different browsers including chrome, chromium and firefox (always used respective cookies)
- different parameters etc.

I'm unable to find another recent/open Issue for Recurbate. As I've tried alot of options and since I'm able to play the video in my browser this must be a bug and not a mistake on my side.

Happy to provide more info if needed. 
Thanks in advance!

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--verbose', '-vU', '--cookies-from-browser', 'firefox', 'https://recurbate.com/play.php?video=60488255']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-79-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.07.22, mutagen-1.46.0, secretstorage-2.3.1, sqlite3-2.6.0, websockets-11.0.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "/home/praes/.mozilla/firefox/wvwksmnk.default-release/cookies.sqlite"
[Cookies] Extracted 374 cookies from firefox
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[Recurbate] Extracting URL: https://recurbate.com/play.php?video=60488255
[Recurbate] 60488255: Downloading webpage
ERROR: [Recurbate] 60488255: This video is only available for registered users; Set your authenticated browser user agent via the --user-agent parameter.. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/recurbate.py", line 29, in _real_extract
    self.raise_login_required(msg=SUBSCRIPTION_MISSING_MESSAGE, method='cookies')
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 1170, in raise_login_required
    raise ExtractorError(msg, expected=True)





```
```

**Comment by bashonly** [comment]
From the error message:

>  Set your authenticated browser user agent via the --user-agent parameter

Did you try doing that? It's not in your log

**Comment by capybaraclique** [comment]
@bashonly 

Yeah, I've tried that one with no success.
Output using the --user agent parameter:

```
[debug] Command-line config: ['--verbose', '-vU', '--cookies-from-browser', 'firefox', '--user-agent', 'firefox', 'https://recurbate.com/play.php?video=60488255']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-79-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.07.22, mutagen-1.46.0, secretstorage-2.3.1, sqlite3-2.6.0, websockets-11.0.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "/home/praes/.mozilla/firefox/wvwksmnk.default-release/cookies.sqlite"
[Cookies] Extracted 374 cookies from firefox
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[Recurbate] Extracting URL: https://recurbate.com/play.php?video=60488255
[Recurbate] 60488255: Downloading webpage
ERROR: [Recurbate] 60488255: This video is only available for registered users; Set your authenticated browser user agent via the --user-agent parameter.. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/recurbate.py", line 29, in _real_extract
    self.raise_login_required(msg=SUBSCRIPTION_MISSING_MESSAGE, method='cookies')
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 1170, in raise_login_required
    raise ExtractorError(msg, expected=True)

```

**Comment by bashonly** [comment]
`firefox` is not a valid user-agent string

it should look like `"Mozilla/5.0 (Windows NT 10.0; rv:118.0) Gecko/20100101 Firefox/118.0"`

but find your browser's *actual* user-agent. you can type "my user-agent" into any major search engine

**Comment by capybaraclique** [comment]
@bashonly 
My bad, just tried the correct user-agent:

```
[debug] Command-line config: ['--verbose', '-vU', '--cookies-from-browser', 'firefox', '--user-agent', 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0', 'https://recurbate.com/play.php?video=60488255']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.8.10 (CPython x86_64 64bit) - Linux-5.15.0-79-generic-x86_64-with-glibc2.29 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.07.22, mutagen-1.46.0, secretstorage-2.3.1, sqlite3-2.6.0, websockets-11.0.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "/home/praes/.mozilla/firefox/wvwksmnk.default-release/cookies.sqlite"
[Cookies] Extracted 374 cookies from firefox
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[Recurbate] Extracting URL: https://recurbate.com/play.php?video=60488255
[Recurbate] 60488255: Downloading webpage
ERROR: [Recurbate] 60488255: This video is only available for registered users; Set your authenticated browser user agent via the --user-agent parameter.. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/recurbate.py", line 29, in _real_extract
    self.raise_login_required(msg=SUBSCRIPTION_MISSING_MESSAGE, method='cookies')
  File "/home/praes/.local/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 1170, in raise_login_required
    raise ExtractorError(msg, expected=True)
```

**Comment by bashonly** [comment]
Have you visited the site in your browser within the last 30 minutes? The site is Cloudflare-protected, it needs fresh cookies from within the last 30 minutes with a matching user-agent

**Comment by bashonly** [comment]
Keeping this issue open to track it for when curl-cffi support is added.

In the meantime, here is a guide for anyone experiencing this issue:
https://github.com/yt-dlp/yt-dlp/wiki/FAQ#im-getting-http-error-403-and-the-site-has-an-open-issue-on-the-tracker-thats-labeled-cloudflare-related-what-can-i-do

**Comment by kevirost** [comment]
ERROR: [facebook] <number>: Cannot parse data ...

I have been following this discussion. I logged into the facebook page, and executed these commands within 30 minutes:

TRIAL 1:

yt-dlp -vU --cookies-from-browser firefox https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522[debug] Command-line config: ['-vU', '--cookies-from-browser', 'firefox', 'https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-6.2.0-26-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, certifi-2020.06.20, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "/home/kevin/.mozilla/firefox/1tngxtjz.default-release-1/cookies.sqlite"
[Cookies] Extracted 2477 cookies from firefox
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[facebook] Extracting URL: https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522
[facebook] 621925246461522: Downloading webpage
[facebook] 621925246461522: Downloading webpage
ERROR: [facebook] 621925246461522: Cannot parse data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/facebook.py", line 707, in _real_extract
    return self._extract_from_url(real_url, video_id)
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/facebook.py", line 654, in _extract_from_url
    raise ExtractorError('Cannot parse data')

TRIAL 2:

yt-dlp -vU --cookies-from-browser firefox --user-agent 'Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0' https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522
[debug] Command-line config: ['-vU', '--cookies-from-browser', 'firefox', '--user-agent', 'Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0', 'https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-6.2.0-26-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, certifi-2020.06.20, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "/home/kevin/.mozilla/firefox/1tngxtjz.default-release-1/cookies.sqlite"
[Cookies] Extracted 2477 cookies from firefox
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[facebook] Extracting URL: https://www.facebook.com/727806382/videos/pcb.10161220512376383/621925246461522
[facebook] 621925246461522: Downloading webpage
[facebook] 621925246461522: Downloading webpage
ERROR: [facebook] 621925246461522: Cannot parse data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/facebook.py", line 707, in _real_extract
    return self._extract_from_url(real_url, video_id)
  File "/home/kevin/.local/lib/python3.10/site-packages/yt_dlp/extractor/facebook.py", line 654, in _extract_from_url
    raise ExtractorError('Cannot parse data')

**Comment by bashonly** [comment]
@kevirost wrong issue
